scope.go

```diff
@@ -478,10 +474,12 @@ func (s *scope) Snapshot() Snapshot {
 }
 
 func (s *scope) Close() error {
-	s.status.Lock()
-	s.status.closed = true
-	close(s.status.quit)
-	s.status.Unlock()
+	if atomic.CompareAndSwapInt32(&s.status.closed, 0, 1) {
+		// prevent any more report loops via tick before forcing report
+		close(s.status.quit)
+		s.reportLoopRun()
+		atomic.StoreInt32(&s.status.closed, 2)
+	}
 
 	if closer, ok := s.baseReporter.(io.Closer); ok {
```

Comment:
If we call close and it's already closed, do we want to close the baseReporter? It's already been closed once no?